nagios_check_cache.py

```diff
@@ -92,13 +92,11 @@ def log_details_and_set_status_for_scenario(self, result_details, feature, prior
         result_details.add("\n")
 
     def set_status_for_scenarios_in_feature(self, feature, priority, result_details):
-        if 'elements' not in feature:
-            log_result_and_exit(0, "OK: Feature %s has no steps at any priority" % feature['id'])
+        if priority != '@normal':
```

Comment:
Could you explain a bit more in the commit message, e.g. how 'priority' is defined now (I assume it defaults to `@normal` somewhere?)

Reply:
I can explain more, but I'll need to clarify what to explain 😄. There's no "default" at the moment: anything without one of these tags won't get picked up by Icinga, was is surprising to me. I could add that in?

Reply:
I'm still confused - if we've removed '@normal' from all the features, why is this line here?

Yes, worth clarifying if a tag is needed in order to be picked up by Icinga :)

> The reason not to is because we can't _completely_ remove it here, until [the corresponding Puppet bit](https://github.com/alphagov/govuk-puppet/blob/4ec2949332079570be5493753bb84a983218a444/modules/icinga/manifests/check_feature.pp) that's using it is gone. We should be able to get rid of *all* mention of "priority", if we do it in stages.

Can we not remove the puppet bit first?

Reply:
I'll try and explain: this script is invoked **by Puppet** with a specific priority. Previously we would compare the specified priority with the tags on each scenario (in the JSON test output), and only report on one that matched. Cucumber has no internal concept of "priority" or "default tag", so any feature without one of our priority tags wasn't picked up by this script and alerted on. This is unnecessarily surprising.

The change here is to remove the part of the script that matches against the priority tag. Puppet is still going to call the script multiple times with different priorities, and the change here means we'll only report back for the time it calls it with the "normal" priority (which the script appends an `@` to).

> Can we not remove the puppet bit first?

We can't remove the Puppet bit because it's priority-based (see [here](https://github.com/alphagov/govuk-puppet/blob/4ec2949332079570be5493753bb84a983218a444/modules/icinga/manifests/check_feature.pp). We need to start ignoring the priority here first (i.e. make the concept redundant) before we can remove it in Puppet.

Does that help?

Reply:
That makes sense, thanks. Can you add that second paragraph to the commit message?

Reply:
Sure, I've rephrased and updated the last commit. Thanks for the questions - I now realise the change is quite difficult to understand unless you're familiar with the script and how/where it's used!

Reply:
Thanks - message looks good except there is a `[1]` that suggests there's a URL missing?

Reply:
Oops - updated 😸